packages/lesswrong/server/search/elastic/ElasticSearchQuery.ts

```diff
@@ -0,0 +1,262 @@
+import type {
+  QueryDslQueryContainer,
+  SearchRequest as SearchRequestInfo,
+} from "@elastic/elasticsearch/lib/api/types";
+import type {
+  SearchRequest as SearchRequestBody,
+} from "@elastic/elasticsearch/lib/api/typesWithBodyKey";
+import {
+  collectionNameToConfig,
+  indexNameToConfig,
+  IndexConfig,
+  Ranking,
+} from "./ElasticSearchConfig";
+import QueryParser, { QueryToken } from "./QueryParser";
+
+/**
+ * There a couple of places where we need a rough origin date
+ * for scaling/faceting/etc. which is defined here. This needn't
+ * be exact - just a date a little older than the oldest searchable
+ * records.
+ */
+export const SEARCH_ORIGIN_DATE = new Date("2014-06-01T01:00:00Z");
```

Comment:
Haven't gotten to where this is used yet, but based on the comment I assume this'll need to be different for LW?